priiloader/source/main.cpp

```diff
@@ -3519,8 +3517,18 @@ int main(int argc, char **argv)
 	flags = GetStateFlags();
 	gprintf("Bootstate %u detected. DiscState %u ,ReturnTo %u & Flags %u & checksum %u\n",flags.type,flags.discstate,flags.returnto,flags.flags,flags.checksum);
 	s8 magicWord = CheckMagicWords();
+	
+	// before anything else, poll input devices such as keyboards to see if we should stop autoboot
+	r = Input_Init();
+	gprintf("Input_Init():%d\n", r );
+	
+	WPAD_SetPowerButtonCallback(HandleWiiMoteEvent);
+	
+	usleep(500000); // Give things enough time to settle down

```

Comment:
this sleep here will make things very slow on startup, can't we only do it when a keyboard is connected for example?

Reply:
I'll see if I it's possible to check if the keyboard is connected before doing that sleep. I'm worried that if there is no sleep than the usb stack won't have had time to even detect a keyboard before we check.

Reply:
Unfortunately it doesn't appear that the usb stack has enough time to recognize the keyboard without a delay. This delay is 0.5s and isn't too bad but it's the only thing that works right now. :/

Reply:
i understand. change the comment to explain its for the KB then. im pretty sure somebody (probably me) will read that in the future and be like "for what to settle down? that's a lot of time" following a delete :P

Reply:
Done